scripts/install.sh

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env bash
+
+# Downloads the latest tarball from https://github.com/amir1376/ab-download-manager/releases and unpacks it into ~/.local/.
+# Creates a .desktop entry for the app in ~/.local/share/applications based on FreeDesktop specifications.
+
+set -euo pipefail
+
+# Check if curl, jq, and tar are installed
+check_dependencies() {
+    for cmd in curl jq tar; do
+        if ! command -v "$cmd" &>/dev/null; then
+            print_message "Error: $cmd is not installed. Please install it and try again."
+            exit 1
+        fi
+    done
+}
+
+# Get the latest version of the app from the GitHub API
+get_latest_version() {
+    curl -s https://api.github.com/repos/amir1376/ab-download-manager/releases/latest | jq -r '.tag_name'
+}
+
+APP_NAME="ABDownloadManager"
+PLATFORM="linux"
+EXT="tar.gz"
+LATEST_VERSION=$(get_latest_version)
+ASSET_NAME=${APP_NAME}_${LATEST_VERSION:1}_${PLATFORM}.${EXT}
+FOLDER_NAME=${APP_NAME}_${LATEST_VERSION}
+BINARY_PATH=$HOME/.local/$FOLDER_NAME/$APP_NAME/bin/$APP_NAME
+
+# Get the download URL for the latest version of the app
+get_download_url() {
+    echo "https://github.com/amir1376/ab-download-manager/releases/download/${LATEST_VERSION}/$ASSET_NAME"
+}
+
+close_if_running() {
+    pid=$(pidof -s "$APP_NAME")
+
+    if [ -n "$pid" ]; then
+        kill -9 "$pid"
+        echo "Closed running instance of AB Download Manager before updating."
+    fi
+}
+
+# Delete the old version of the app if it exists
+delete_old_version() {
+    local installed_dir
+    close_if_running
+    installed_dir=$(find "$HOME/.local" -maxdepth 1 -type d -name "ABDownloadManager_*" | head -n 1)
+    if [ -n "$installed_dir" ]; then
+        rm -rf "$installed_dir"
+        rm -rf "$HOME/.local/bin/$APP_NAME"
+        echo "Old version of AB Download Manager deleted"
+    fi
+}
+
+# Get the installed version of the app if it exists
+get_installed_version() {
```

Comment:
I added the `ABDownloadManager --version` cli command which returns the current version of the app
https://github.com/amir1376/ab-download-manager/commit/0dc4c676ee83083430a65048bf151aa47a9be55a

Reply:
Pushed new commit with the intended changes